File: app/src/main/java/com/sjn/stamp/ui/DrawerMenu.java
package com.sjn.stamp.ui;

import android.support.v4.app.Fragment;

import com.sjn.stamp.R;
import com.sjn.stamp.ui.fragment.SettingFragment;
import com.sjn.stamp.ui.fragment.media_list.AllSongPagerFragment;
import com.sjn.stamp.ui.fragment.media_list.QueueListFragment;
import com.sjn.stamp.ui.fragment.media_list.RankingPagerFragment;
import com.sjn.stamp.ui.fragment.media_list.StampPagerFragment;
import com.sjn.stamp.ui.fragment.media_list.TimelineFragment;

public enum DrawerMenu {
    HOME(R.id.navigation_home) {
        @Override
        public Fragment getFragment() {
            return new AllSongPagerFragment();
        }
    },
    TIMELINE(R.id.navigation_timeline) {
        @Override
        public Fragment getFragment() {
            return new TimelineFragment();
        }
    },
    QUEUE(R.id.navigation_queue) {
        @Override
        public Fragment getFragment() {
            return new QueueListFragment();
        }
    },
    STAMP(R.id.navigation_stamp) {
        @Override
        public Fragment getFragment() {
            return new StampPagerFragment();
        }
    },
    RANKING(R.id.navigation_ranking) {
        @Override
        public Fragment getFragment() {
            return new RankingPagerFragment();
        }
    },
    SETTING(R.id.navigation_setting) {
        @Override
        public Fragment getFragment() {
            return new SettingFragment();
        }
    },;
    final int mMenuId;

    DrawerMenu(int menuId) {
        mMenuId = menuId;
    }

    public int getMenuId() {
        return mMenuId;
    }

    public abstract Fragment getFragment();

    public static DrawerMenu of(long menuId) {
        for (DrawerMenu drawerMenu : DrawerMenu.values()) {
            if (drawerMenu.mMenuId == menuId) return drawerMenu;
        }
        return null;
    }

    public static DrawerMenu first() {
        return HOME;
    }
}

File: app/src/main/java/com/sjn/stamp/ui/DrawerMenu.kt
package com.sjn.stamp.ui

import android.support.v4.app.Fragment
import com.sjn.stamp.R
import com.sjn.stamp.ui.fragment.SettingFragment
import com.sjn.stamp.ui.fragment.media_list.*

enum class DrawerMenu(val menuId: Int) {
    HOME(R.id.navigation_home) {
        override val fragment: Fragment
            get() = AllSongPagerFragment()
    },
    TIMELINE(R.id.navigation_timeline) {
        override val fragment: Fragment
            get() = TimelineFragment()
    },
    QUEUE(R.id.navigation_queue) {
        override val fragment: Fragment
            get() = QueueListFragment()
    },
    STAMP(R.id.navigation_stamp) {
        override val fragment: Fragment
            get() = StampPagerFragment()
    },
    RANKING(R.id.navigation_ranking) {
        override val fragment: Fragment
            get() = RankingPagerFragment()
    },
    SETTING(R.id.navigation_setting) {
        override val fragment: Fragment
            get() = SettingFragment()
    };

    abstract val fragment: Fragment

    companion object {

        fun of(menuId: Long): DrawerMenu? = DrawerMenu.values().firstOrNull { it.menuId.toLong() == menuId }

        fun first(): DrawerMenu = HOME
    }
}
